modeshape-jcr/src/main/java/org/modeshape/jcr/query/engine/ScanningQueryEngine.java

```diff
@@ -1537,6 +1538,9 @@ protected RowFilter createFilter() {
                     return new DynamicOperandFilter(operation) {
                         @Override
                         protected boolean evaluate( Object leftHandValue ) {
+                            if (Path.class.isAssignableFrom(defaultType.getType())) {
+                                leftHandValue = pathFactory.create(leftHandValue);
```

Comment:
According to [the API](https://github.com/ModeShape/modeshape/blob/master/modeshape-jcr/src/main/java/org/modeshape/jcr/value/ValueFactory.java#L472) it should.